### PR TITLE
Mock data & test update

### PR DESCRIPTION
**Pull request by solidoracle**
### Overview

Constructed mock data to run the reward distribution scripts when the .env variable ETHEREUM_NETWORK is different from ‘mainnet’. I also built a test that now verifies the weights that we are assigning to each voter.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,7 +35,9 @@ In `.env` set the VOTE_ID string to the Sarco DAO vote number.
 
 ## Reward Distribution Script Execution
 
-In `.env` set your INFURA_API_KEY and TOTAL_REWARDS_WEI as the amount to distribute as rewards. To simulate the actual distribution, set TOTAL_REWARDS_WEI to the unallocated rewards of the Collection contract.
+For mainnet in `.env` set ETHEREUM_NETWORK="mainnet" as well as your INFURA_API_KEY and TOTAL_REWARDS_WEI as the amount to simulate rewards distribution. To simulate the actual reward distribution mechanics, set TOTAL_REWARDS_WEI to the unallocated rewards of the Collection contract.
```
Comment (DarksightKellar) [inline README.md]: Could you update this doc to more clearly separate `mainnet` and `simulation`? Definitely don't want to suggest in any way shape or form that the `mainnet` may be used for any sort of simulation

File: README.md [hunk 2]
```diff
@@ -54,18 +56,24 @@ npm run deploy
...
+
 The script is executed on the deployed Sarco Collection Contract, so its address needs to be added in `.env` COLLECTION_CONTRACT_ADDRESS.
 
-Any unallocated rewards in the Collection Contract will be distributed as rewards to the voters of the selected VOTE_ID in `.env`.
+When on mainnet, any unallocated rewards in the Collection Contract will be distributed as rewards to the voters of the selected VOTE_ID in `.env`.
 
-To distribute any unallocated rewards from the Collection Contract run:
+For test environment set in `.env` set ETHEREUM_NETWORK to an empty string.
+
+To conole log the allocation in the terminal run:
```
Comment (DarksightKellar) [inline README.md]: Small typo here @ "conole"

File: README.md [hunk 3]
```diff
@@ -54,18 +56,24 @@ npm run deploy
...
 
 ```
 npm run distribution
 ```
 
 ## Execute Test
 
+For testing with mock data set in `.env` ETHEREUM_NETWORK to an empty string, then run:
```
Comment (DarksightKellar) [inline README.md]: Probably not terribly important, but does the codebase not account for running tests on testnets like goerli? In which case the mock data would be gotten from equivalent testnets? 

Something to consider, maybe, but definitely not for this PR

File: test/index.ts [hunk 4]
```diff
@@ -2,6 +2,8 @@ import { expect } from 'chai';
 import { calculateRewardsAmounts, zero, Reward } from '../src/index';
 import { describe } from 'mocha';
 import { BigNumber, ethers } from 'ethers';
+import { mockStakingAmount, totalStakingAmount } from '../src/mocks/mock-data';
+require('dotenv').config();
```
Comment (DarksightKellar) [inline test/index.ts]: Sorry minor nitpick here, but mind using the `import` syntax here instead of `require`?

File: test/index.ts [hunk 5]
```diff
@@ -14,11 +16,27 @@ function getSum(distributions: Reward[]): BigNumber {
 }
 
 describe('Script: rewards distribution ', () => {
-  it('Sum distributed to voters should be equal to initial amount set to distribute', async () => {
-    const TOTAL_REWARDS_AMOUNT = ethers.utils.parseEther('100');
-    const rewardsObject = await calculateRewardsAmounts(TOTAL_REWARDS_AMOUNT);
-    const rewardsSum = getSum(rewardsObject);
+  if (process.env.ETHEREUM_NETWORK === 'mainnet') {
```
Comment (DarksightKellar) [inline test/index.ts]: The tests under this condition are pointless IMO. To my previous point about using mainnet -- we don't want to be running tests with unpredictable, live data, especially not on mainnet. 

Even if it works (I've heard some companies do it lol), it shouldn't be encouraged at all.

File: test/index.ts [hunk 6]
```diff
@@ -14,11 +16,27 @@ function getSum(distributions: Reward[]): BigNumber {
...
+    it('Sum distributed to voters should be equal to initial amount set to distribute', async () => {
+      const TOTAL_REWARDS_AMOUNT = ethers.utils.parseEther('100');
+      const rewardsObject = await calculateRewardsAmounts(TOTAL_REWARDS_AMOUNT);
+      const rewardsSum = getSum(rewardsObject);
+      expect(Number(rewardsSum)).closeTo(Number(TOTAL_REWARDS_AMOUNT), 1000000);
+    });
+  } else {
+    it('Reward per voter should be distributed as per relative weight of their Sarco-VR balance', async () => {
```
Comment (DarksightKellar) [inline test/index.ts:21]: Yep this is what we want to keep. Thanks for this. 

There just might be other edge case tests we could add, but I guess the whole functionality is straightforward and intuitive enough that we should be good for now.

Or, well, there's also the argument that you gotta be super thorough when dealing with people's funds.

Thoughts? @sethhrbek @katien @knoll3 

I personally can't think of anything else to test, apart from maybe testing for large number of accounts?

**Review by DarksightKellar — CHANGES_REQUESTED** [review]
Looks good! I had a couple comments on some changes, and then we can get this merged in

File: test/index.ts (hunk 5 above)
Comment (solidoracle) [reply test/index.ts]: Ok I will condition the testing to only if process.env.ETHEREUM_NETWORK != 'mainnet'

File: README.md (hunk 3 above)
Comment (solidoracle) [reply README.md]: This is not accounted for atm. I've set up tests to run with the mock data I generated when process.env.ETHEREUM_NETWORK != 'mainnet'

File: README.md (hunk 1 above)
Comment (solidoracle) [reply README.md]: Absolutely - I divided the ReadMe to make the distinction more clear

**Review by DarksightKellar — APPROVED** [review]
